src/server/routes/oauth.ts

```diff
@@ -270,6 +270,19 @@ const authenticationHandler = async (
 			return res.redirect(303, authState.queryParams.fromURI);
 		}
 
+		// temporary fix: if the user registered on the app (the token will be prefixed),
+		// and instead of ending up back in the app but in a mobile browser instead,
+		// for now we simply redirect them to the default return url
+		// which is normally the guardian home page, by setting authState.confirmationPage to undefined
+		// this will be fixed when we either use a deep link with a custom scheme, or passwordless OTP flow
+		if (
+			authState.data?.hasAppPrefix &&
+			authState.confirmationPage === consentPages[0].path
```

Comment:
I'm a bit confused on the logic here - why do we need to check if `confirmationPage` is the same as the first path in the `consentPages` array? Would it not be more futureproof if we just check that `confirmationPage` matches _any_ of the `consentPages` paths?

Reply:
`consentPages[0].path` is currently a pattern we use elsewhere in gateway (unfortunately), so I decided to use that. But I'll update it so that it could match either.

Reply:
Updated!